src/pkg/bundle/create.go

```diff
@@ -107,3 +114,69 @@ func (b *Bundle) confirmBundleCreation() (confirm bool) {
 	}
 	return true
 }
+
+// processValuesFiles reads values from valuesFiles and updates the bundle with the override values
+func (b *Bundle) processValuesFiles() error {
+	// Populate values from valuesFiles if provided
+	for i, pkg := range b.bundle.Packages {
+		for componentName, overrides := range pkg.Overrides {
+			for chartName, bundleChartOverrides := range overrides {
+				valuesFilesToMerge := make([][]types.BundleChartValue, 0)
+				// Iterate over valuesFiles in reverse order to ensure subsequent value files takes precedence over previous ones
+				for _, valuesFile := range bundleChartOverrides.ValuesFiles {
+					// Check relative vs absolute path
+					fileName := filepath.Join(b.cfg.CreateOpts.SourceDirectory, valuesFile)
+					if filepath.IsAbs(valuesFile) {
+						fileName = valuesFile
+					}
+					// read values from valuesFile
+					values, err := chartutil.ReadValuesFile(fileName)
+					if err != nil {
+						return err
+					}
+					if len(values) > 0 {
+						// populate BundleChartValue slice to use for merging existing values
+						valuesFileValues := make([]types.BundleChartValue, 0, len(values))
+						for key, value := range values {
+							valuesFileValues = append(valuesFileValues, types.BundleChartValue{Path: key, Value: value})
+						}
+						valuesFilesToMerge = append(valuesFilesToMerge, valuesFileValues)
+					}
+				}
+				override := b.bundle.Packages[i].Overrides[componentName][chartName]
+				// add override values to the end of the list of values to merge since we want them to take precedence
+				valuesFilesToMerge = append(valuesFilesToMerge, override.Values)
+				override.Values = mergeBundleChartValues(valuesFilesToMerge...)
+				b.bundle.Packages[i].Overrides[componentName][chartName] = override
+			}
+		}
+	}
+	return nil
+}
+
+// mergeBundleChartValues merges lists of BundleChartValue using the values from the last list if there are any duplicates
+// such that values from the last list will take precedence over the values from previous lists
+func mergeBundleChartValues(bundleChartValueLists ...[]types.BundleChartValue) []types.BundleChartValue {
+	merged := make([]types.BundleChartValue, 0)
+	paths := make(map[string]bool)
+
+	// Iterate over each list in order
+	for _, bundleChartValues := range bundleChartValueLists {
+		// Add entries from the current list to the merged list, overwriting any existing entries
+		for _, bundleChartValue := range bundleChartValues {
+			if _, ok := paths[bundleChartValue.Path]; ok {
+				// Remove the existing entry from the merged list
+				for i, mergedValue := range merged {
+					if mergedValue.Path == bundleChartValue.Path {
+						merged = append(merged[:i], merged[i+1:]...)
```

Comment:
Pretty sure the code will panic if `[i+1]` goes over the slice's len. Couple of options here:
1. Process these values as a map instead of a list and convert the result to a list at the end
2. Can potentially use `helpers.MergeSlices`

Reply:
🤖 can gave me the exact same logic the other day when I was de-duping lists, and I ended up getting panics during tests

Reply:
❤️ the variadic args!

Reply:
lol :robot: doubled down, and then apologized. Went with option 1. map -> slice